### PR TITLE
feat: Add Sigstore verification capabitilies in PolicyServers CRD

### PR DESCRIPTION
**Pull request by viccuad**
- feat: Add a new key to PolicyServer spec:
  - `verificationConfig`: Name of k8s configmap in the same ns that contains
    the verification configuration for the policy-server[1], in the `verification-config` key.

  If verification enabled, the configmap would be mounted on the containers of the policy-server, and it will be used to enable verification with that configuration.

- refactor!: Rename functions and variables related to the CA secrets as such, the disambiguate between the verification secret and the CA secrets. This renames condition `PolicyServer{,CA}SecretReconciled`, which is a breaking change for users.
- tests: Unit tests are covered by existing ones, as there's change of env in the container.
- tests: A simple E2E idempotent test has been added, see `e2e-tests` directory.
  I'm happy dropping it, but it was easier to push that than to paste and explain the testing in the PR. They may need more setup and cleanup if more tests are added, but should be doable.
  
  I am on the opinion that these kind of tests should be part of the repo, as they are tightly coupled with the code that implements them (and aid review of features). A CI engine, living in a different repo, can be used as a dependency to run these kubewarden-controller e2e tests (and others, with more coverage of the full stack, for example chart behavior tests, living in the chart repo). That way, we can check in the tests closer to the code, for easier refactoring and debugging of failing tests.
  
  
Fixes: https://github.com/kubewarden/kubewarden-controller/issues/137
Depends on https://github.com/kubewarden/policy-server/pull/132, so until that is merged, one needs to build a local policy-server image, inject it in cluster, and change `e2e-tests/test-verification/resources/server-with-verification.yml` to use it (plus change ImagePullPolicy to Never on the server deployment).

### CODE REVIEW

File: apis/policies/v1alpha2/policyserver_types.go [hunk 1]
```diff
@@ -73,12 +83,15 @@ const (
...
+	// PolicyServerCASecretReconciled represents the condition of the
 	// Policy Server Secret reconciliation
-	PolicyServerSecretReconciled PolicyConditionType = "PolicyServerSecretReconciled"
+	PolicyServerCASecretReconciled PolicyConditionType = "PolicyCAServerSecretReconciled"
 	// PolicyServerCARootSecretReconciled represents the condition of the
 	// Policy Server CA Root Secret reconciliation
 	PolicyServerCARootSecretReconciled PolicyConditionType = "PolicyCARootServerSecretReconciled" // nolint:gosec
+	// PolicyServerVerificationSecretReconciled represents the condition of the
+	// Policy Server Verification Secret reconciliation
+	// PolicyServerVerificationSecretReconciled PolicyConditionType = "PolicyServerVerificationSecretReconciled"
```
Comment (flavio) [inline apis/policies/v1alpha2/policyserver_types.go:94]: Are you sure this line should be commented? :thinking:

**Review by flavio — COMMENTED** [review]
Looks good to me, I just left one comment

**Review by flavio — APPROVED** [review]
LGTM, but this needs to be rebased against main

**Comment by viccuad** [comment]
Rebased onto main; I would still wait until the policy-server has the functionality, so one can run the e2e-tests on this PR manually.

**Comment by flavio** [comment]
> Rebased onto main; I would still wait until the policy-server has the functionality, so one can run the e2e-tests on this PR manually.

Fine with me :+1:

**Review by jvanz — APPROVED** [review]

**Comment by flavio** [comment]
> Rebased onto main; I would still wait until the policy-server has the functionality, so one can run the e2e-tests on this PR manually.

JFYI: the latest policy server from main has the sigstore integration enabled

**Comment by viccuad** [comment]
Run the e2e-tests included here, and it works. Discovered https://github.com/kubewarden/policy-server/pull/141 though.

Still, I realized that we have bad UX:
We are using a secret for the `verification.yml` config. From what I can tell, there's no way to mount a generic secret under a filename, as it mounts keys inside the `data` of the secret instead. In our case, by how we create the secret in the tests using `--from-file`, we have a key `data.verification.yml`, as the filename for the secret is named `verification.yml`.

This is confusing, and users may not realize that they need to name their file `verification.yml` (nor they may want to).

I think verification.yml should come from a Configmap, which can then be mounted as a file.

**Comment by viccuad** [comment]
I have redone the PR with a configmap for the verification config, instead of a secret. I have force-pushed changing the commit "feat: Add verification options to PolicyServer " incluse and onwards, sorry WRT to review (ended up with too many fixups, preferred to squash).

Tests passing locally.

Both Configmaps and Secrets can't be mounted as 1 file. Hence, when creating the configmap, one needs to specify that the key `verification-config` is the one that will be used. Example from the tests:
> kubectl create configmap my-verification-config \
   --from-file=**verification-config**=./resources/my-verification-config.yml \
   --namespace=kubewarden

**Review by flavio — APPROVED** [review]
LGTM

**Comment by viccuad** [comment]
Force-pushed again, changing the last 2 commits, so the PolicyServer CRD only gains the new setting `verificationConfig`. Updated the e2e-tests example `server-with-verification.yml`.

Tests passing locally.

**Review by flavio — CHANGES_REQUESTED** [review]
I think there's one minor change to be done, then this is going to be good for merge

File: internal/pkg/admission/policy-server-deployment.go [hunk 2]
```diff
@@ -238,6 +240,27 @@ func (r *Reconciler) deployment(configMapVersion string, policyServer *policiesv
...
+				ReadOnly:  true,
+				MountPath: constants.PolicyServerVerificationConfigContainerPath,
+			},
+		)
+		admissionContainer.Env = append(admissionContainer.Env,
+			corev1.EnvVar{
+				Name:  "KUBEWARDEN_ENABLE_VERIFICATION",
+				Value: "yes",
+			},
+		)
```
[the remark this answers is not in the document]
Comment (viccuad) [reply internal/pkg/admission/policy-server-deployment.go]: Done in https://github.com/kubewarden/kubewarden-controller/pull/139/commits/67c811182db60cf351783a0e979612b76c99d995!

**Review by flavio — APPROVED** [review]
LGTM

**Pull request merged**